build.gradle.kts

```
@@ -211,6 +211,12 @@ tasks {
 		options.compilerArgs.add("-Werror")
 		// Do not break the build on "exports" warnings (see CONTRIBUTING.md for details)
 		options.compilerArgs.add("-Xlint:all,-exports")
+
+		if(project.version != "unspecified") {
+			// Add version to Java modules
+			options.javaModuleVersion.set(project.version.toString());
+		}
+
```

Formatting nitpicks:

```suggestion
		if (project.version != "unspecified") {
			// Add version to Java modules
			options.javaModuleVersion.set(project.version.toString());
		}
```